benchmark/Benchmark.h

```diff
@@ -40,58 +42,113 @@ class BenchmarkRecords {
+    // The order of creation for the entries of recordGroups_.
+    std::vector<std::string> recordGroupsOrder_;
   
     // A hash map of all the created RecordTables. For faster access.
     // The key for a RecordTable is it's descriptor.
-    ad_utility::HashMap<std::string, RecordTable> _recordTables;
-    
+    ad_utility::HashMap<std::string, RecordTable> recordTables_;
+     // The order of creation for the entries of recordTables_.
+    std::vector<std::string> recordTablesOrder_;
+
     /*
      * @brief Return execution time of function in seconds.
+     *
+     * @tparam FUNCTION_TYPE Best left to type inference.
+     *
+     * @param functionToMeasure Must be a function, or callable.
+     */
+    template<typename FUNCTION_TYPE>
+    float measureTimeOfFunction(const FUNCTION_TYPE& functionToMeasure) const{
+      ad_utility::Timer benchmarkTimer;
+         
+      benchmarkTimer.start();
+      functionToMeasure();
+      benchmarkTimer.stop();
+
+      return benchmarkTimer.secs();
+    }
+
+    /*
+     * @brief Translate the given hash map keys to their values and return
+     * those values in the same order as the keys.
+     *
+     * @tparam MAP_KEY_TYPE, MAP_VALUE_TYPE The types for the (key, value)
+     *  pairs of the hash map.
+     *
+     * @param hashMap The hash map, where the values should be looked up in.
+     * @param hashMapKeys The keys, for which the values should be looked up
+     *  and inserted in a vector.
+     *
+     * @returns A vector of the hashMap[hashMapKeys] values, in the same order
+     *  as the keys in hashMapKeys.
      */
-    float measureTimeOfFunction(const std::function<void()>& functionToMeasure) const;
+    template<typename MAP_KEY_TYPE, typename MAP_VALUE_TYPE>
+    static const std::vector<MAP_VALUE_TYPE> createVectorOfHashMapValues(
+        const ad_utility::HashMap<MAP_KEY_TYPE, MAP_VALUE_TYPE>& hashMap,
+        const std::vector<MAP_KEY_TYPE>& hashMapKeys);
 
   public:
 
     /*
      * @brief Measures the time needed for the execution of the given function and
      * saves it, together with a description, in a normal list.
      *
+     * @tparam FUNCTION_TYPE Best left to type inference.
+     *
      * @param descriptor A description, of what kind of benchmark case is
      *  getting measured. Needed, because otherwise nobody would be able to
      *  tell, which time corresponds to which benchmark.
      * @param functionToMeasure The function, that represents the benchmark.
      *  Most of the time a lambda, that calls the actual function to benchmark
      *  with the needed parameters.
      */
-    void addSingleMeasurment(const std::string descriptor,
-        const std::function<void()>& functionToMeasure);
+    template<typename FUNCTION_TYPE>
+    void addSingleMeasurement(const std::string& descriptor,
+        const FUNCTION_TYPE& functionToMeasure) {
+      singleMeasurements_.push_back(RecordEntry{descriptor,
```

Comment:
Same template constraint as above, and also call it `Function`.

Reply:
The constraints were already implemented and I changed the name in [8009ba3](https://github.com/ad-freiburg/qlever/pull/860/commits/8009ba3301f0cff3fa8441b2dd94f7785ce57fb6).